### PR TITLE
APPS-4242 add glue tests

### PR DESCRIPTION
**Pull request by sergeyspryker**
https://release.spryker.com/release-groups/view/4171

### CODE REVIEW

File: src/Spryk/Model/Spryk/Executor/SprykExecutor.php [hunk 1]
```diff
@@ -72,28 +71,36 @@ class SprykExecutor implements SprykExecutorInterface
      */
     protected int $currentExecutionLevel = 0;
 
+    /**
+     * @var \SprykerSdk\Spryk\Model\Spryk\Executor\ConditionMatcher\ConditionMatcherInterface
+     */
+    private $conditionMatcher;
```
Comment (stereomon) [inline src/Spryk/Model/Spryk/Executor/SprykExecutor.php:77]: Please always make use of typed properties and use protected.

File: src/Spryk/Model/Spryk/Filter/HttpResponseCodeToConstantNameFilter.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+use ReflectionClass;
+use RuntimeException;
+use Symfony\Component\HttpFoundation\Response;
+
+class HttpResponseCodeToConstantNameFilter implements FilterInterface
+{
+    /**
+     * @var array<string, mixed>
+     */
+    protected $responseClassConstants = [];
```
Comment (stereomon) [inline src/Spryk/Model/Spryk/Filter/HttpResponseCodeToConstantNameFilter.php:19]: property type is missing. Shouldn't this be a static one for performance reasons?

File: tests/SprykerSdkTest/Spryk/Model/Spryk/Executor/ConditionMatcherTest.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+use SprykerSdk\Spryk\Model\Spryk\Executor\ConditionMatcher\ConditionMatcher;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\ExpressionLanguage\SyntaxError;
+
+class ConditionMatcherTest extends Unit
+{
+    /**
+     * @var \SprykerSdk\Spryk\Model\Spryk\Executor\ConditionMatcher\ConditionMatcher
+     */
+    protected $conditionMatcher;
```
Comment (stereomon) [inline tests/SprykerSdkTest/Spryk/Model/Spryk/Executor/ConditionMatcherTest.php:22]: Property type is missing

File: tests/SprykerSdkTest/Spryk/Model/Spryk/Executor/ConditionMatcherTest.php [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    /**
+     * @return void
+     */
+    public function setUp(): void
+    {
+        parent::setUp();
```
Comment (stereomon) [inline tests/SprykerSdkTest/Spryk/Model/Spryk/Executor/ConditionMatcherTest.php:29]: I personally don't like to use this approach. What's against using `\SprykerSdkTest\Module\SprykHelper::getClass()` or adding your own method to get this service from the tester as done with e.g. `\SprykerSdkTest\Module\SprykHelper::getClassDumper()`?

File: tests/SprykerSdkTest/Spryk/Model/Spryk/Filter/HttpResponseCodeToConstantNameFilterTest.php [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+    /**
+     * @return void
+     */
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        if ($this->filter === null) {
+            $this->filter = new HttpResponseCodeToConstantNameFilter();
```
Comment (stereomon) [inline tests/SprykerSdkTest/Spryk/Model/Spryk/Filter/HttpResponseCodeToConstantNameFilterTest.php:28]: Tests also must make use of the DependencyInjection container.

**Review by stereomon — CHANGES_REQUESTED** [review]

File: config/spryk/spryks/Spryker/Glue/Test/AddGlueTestTester.yml [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        value: "tests/{{ organization }}Test/{{ application }}/{{ module }}/_support"
+
+postSpryks:
+    - AddMethod:
+        condition: "(httpMethod === 'patch' || httpMethod === 'get') && isBulk === false"
+        arguments:
+            httpMethod:
+                inherit: true
+            method:
+                value: "seeResponseJsonContains{{ zedDomainEntity }}IdWithUuid"
```
[the remark this answers is not in the document]
Comment (sergeyspryker) [reply config/spryk/spryks/Spryker/Glue/Test/AddGlueTestTester.yml:57]: But we check Id and Uuid

File: config/spryk/templates/Glue/Test/GlueTestApiCest/ApiCest.php.twig [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+     * @param \{{ organization }}Test\{{ application }}\{{ resource | normalizeResourceName | ucfirst }}{{ applicationType }}Api\{{ resource | normalizeResourceName | ucfirst }}{{ applicationType }}ApiTester $I
+     *
+     * @return void
+     */
+    public function loadFixtures(\{{ organization }}Test\{{ application }}\{{ resource | normalizeResourceName | ucfirst }}{{ applicationType }}Api\{{ resource | normalizeResourceName | ucfirst }}{{ applicationType }}ApiTester $I): void
+    {
+        /** @var \{{ organization }}Test\{{ application }}\{{ module }}\{{ suite }}\Fixtures\{{ fixtureClass }} $fixtures */
+        $fixtures = $I->loadFixtures({{ fixtureClass }}::class);
+
+        $this->fixtures = $fixtures;
```
[the remark this answers is not in the document]
Comment (sergeyspryker) [reply config/spryk/templates/Glue/Test/GlueTestApiCest/ApiCest.php.twig:36]: Was added due to return value typeHint

**Review by stereomon — APPROVED** [review]

**Pull request merged**